### DISCUSSION TITLE
Limit table records based on filter

### DISCUSSION THREAD

**Discussion by Didier-devfr** [filamentphp/filament#13128] [Help]
### Package

Table builder

### Package Version

v3.2.85

### How can we help you?

Hello guys,
I have a resource with thousands of results.
I would like to limit the number of results in my table based on a filter, I have a checkbox that displays an input number with my limit value, for example 200.
But I would like to keep a pagination of 10 results per page and therefore display a maximum of 20 pages.
Currently the only solution I have found is to disable pagination, but this causes performance issues.
Is this possible? Thanks!

**Comment by dave-mills** [comment]
I don't think this is possible without rewriting how the table handles pagination. When constructing the Eloquent query, `->paginate()` overrides any `->limit()` or `->offset()` values, so even if you add a filter to modify the base query on the table, it won't have any effect - unless you disable the default pagination, as you've found. 

I imagine there's a way to make your own pagination following the guides in the [docs](https://laravel.com/docs/11.x/pagination#manually-creating-a-paginator), but that sounds pretty complex!

**Reply by Didier-devfr** [reply]
Ok thank you for your answer, I will look at the docs for more info !

**Comment by Didier-devfr** [comment]
I almost succeeded... I added a counter for each result line and then I added a where to retrieve only the first x results, here is part of the code:
```
Tables\Filters\Filter::make('campaign')
  ->form([
    Forms\Components\Checkbox::make('is_campaign'),
    Forms\Components\TextInput::make('sms_credit'),
  ])
  ->baseQuery(function (Builder $query, array $data) {
    $query
      ->from(DB::raw('(SELECT @row_number := 0) AS t, interimaires'));
    if ($data['is_campaign']) {
      $query->whereRaw('(@row_number := @row_number + 1) <= ?', [(int)$data['sms_credit']]);
    }
  })
```
my last problem is that the more my value increases, the more I'm losing results, here are some examples of values ​​and the number of results returned:
80 => 80
200 => 199
900 => 897
1000 => 990
if anyone has a solution to always get the exact results that would be perfect!
